cogs/moderation.py

```diff
@@ -492,6 +572,184 @@ async def on_member_kick(self, target, entry):
+                "Emergency Staff Alert role not found for this guild. Please ask an Administrator to set one up.",
+                ephemeral=True,
+            )
+
+        number_staff = len(role.members)
+        confirm_embed = discord.Embed(
+            color=discord.Color.red(),
+            title="🚨 Emergency Staff Alert",
+            description=textwrap.dedent(
+                f"""
+                This command is designed for use in case of emergencies actively happening in our server(s) that need immediate staff attention. This will ping **{number_staff}** staff member{'' if number_staff == 1 else 's'} currently assigned to the {role.mention} role, and you will be assisted shortly.
+                """
+            ),
+        )
+        confirm_embed.add_field(
+            name="Are you sure that you want to send an Emergency Staff Alert for the following reason?",
+            value=reason,
+            inline=False,
+        )
+
+        rules_embed = discord.Embed(
+            color=discord.Color.blurple(),
+            title="Use Cases & Abuse",
+            description=textwrap.dedent(
+                f"""
+                Abuse of this alert system is **strictly prohibited** and **will** result in repercussions if used maliciously. Here are some examples to help understand when and when not to use it, this is not exhaustive:
+                **✅ Acceptable Cases**
+                - Sending NSFW/disturbing content in our server(s)/DMs
+                - Advertising Crosstrading/Distribution of automated scripts in our server(s) that violate our ToS
+                - Malicious/excessive spam in our server(s)
+                - Advertising links to malicious/scam websites in our server(s)/DMs
+                - Extreme Toxicity/Harassment/Trolling
+                - Actively violating any other rule to an excessive extent
+                **<:white_cross_mark:1193650425166045224> Unacceptable Cases**
+                - Suspected autocatching in our server(s)
+                - Server advertisement
+                - Bot outages/bugs/glitches — Please use #bug-reports or ping a Developer in case of emergency
+                - Asking staff to check appeals/applications
+                """
+            ),
+        )
+        rules_embed.set_footer(text="Please use `?report` in unacceptable cases that violate our rules.")
+
+        if not await ctx.confirm(timeout=120, embeds=[confirm_embed, rules_embed], ephemeral=True):
+            ctx.command.reset_cooldown(ctx)
+            return await ctx.send("Aborted.", ephemeral=True)
+
+        alert_embed = discord.Embed(color=discord.Color.red(), title="🚨 Emergency Staff Alert Issued", description="")
+        alert_embed.set_author(
+            name=f"{ctx.author} ({ctx.author.id})",
+            icon_url=ctx.author.display_avatar,
+        )
+        alert_embed.add_field(name="Reason", value=reason, inline=False)
+        view = EmergencyView(ctx)
+        view.add_item(
```

Comment:
can't this button be part of the class?

Reply:
To my knowledge the only ways are to either 1) `clear_items()` in the `__init__` and then add the buttons manually 2) Have a separate button subclass for the resolve button and then add them in the `__init__` manually. This way felt easier to me definitely but can do one of those two (or any other way i may be unaware of), what do you think?